R/calcFeDemandONONSPEC.R

```diff
@@ -0,0 +1,112 @@
+#' Calculate historic and projected other non-specified energy demand
+#'
+#' Project the IEA flow ONONSPEC into the future.
+#'
+#' @author Robin Hasse
+#'
+#' @returns list with MagPIE object
+#'
+#' @importFrom madrat toolGetMapping calcOutput
+#' @importFrom magclass getItems getSets<-
+#' @importFrom dplyr %>% .data filter group_by across all_of group_modify
+#'   slice_tail ungroup reframe
+
+calcFeDemandONONSPEC <- function() {
+
+  # project yearly until
+  endOfProjection <- 2100
+
+  # approach constant level after ... years
+  yearsUntilConst <- c(15, 20, 25)
+
+  # ratio of EOH slope that qualifies as constant
+  constTolerance <- 0.05
+
+
+
+
+  # FUNCTIONS ------------------------------------------------------------------
+
+  .getHistFlows <- function() {
+    data <- calcOutput("IOEdgeBuildings",
+                       subtype = "output_EDGE",
+                       aggregate = FALSE)
+    getSets(data) <- c("region", "period", "flow")
+    # ONONSPEC have been mapped to feoth* before
+    itemsONONSPEC <- grep("^feoth.+$", getItems(data, 3), value = TRUE)
+    data[, , itemsONONSPEC]
+  }
+
+
+  .getEOHcoefs <- function(x, key) {
+    m <- lm(value~period, x)
+    periodEOH <- max(x$period)
+    coefs <- data.frame(periodEOH = periodEOH,
+                        slopeEOH = coef(m)[["period"]],
+                        valueEOH = predict(m, list(period = periodEOH)))
```

Comment:
```suggestion
                        valueEOH = predict(m, data.frame(period = periodEOH)))
```
Not sure if this makes much of a difference, but usually ```predict``` takes data frames.

Reply:
It works with both and I used a list here as I only pass one number.